index.js

```diff
@@ -15,6 +15,7 @@ const streamToBuffer = require('fast-stream-to-buffer')
 const StreamChopper = require('stream-chopper')
```

Comment:
👍 to the changes in this file.  I see we're assigning our new logger or a NoopLogger, and have added some internal properties for tracking metrics related to the client with a `_getStats` helper function.  The simpler approach makes sense to me for tracking agent perf at this point (vs. a formal metrics system like Measured, which can be heavy).  I also see we've renamed the `_received` property to the more precise `_numEventsEnqueued` and renamed `_writevCleaned` to `_writeBatch`.